README.dev.md

```diff
@@ -6,11 +6,612 @@ You will need a machine with at least 2GB of RAM available (see
+- runs the `provision.py` script, which downloads all required dependencies
+  and sets up the python environment for Zulip within the virtual
+  machine/container. 
+
+You will need an active internet connection during the entire processes.
+
+Once `vagrant up` has completed, connect to the dev environment with `vagrant
+ssh`:
+
+```
+christie@win10 ~/zulip
+$ vagrant ssh 
+```
+
+You should see something like this on Windows and OS X:
+
+```
+Welcome to Ubuntu 14.04.4 LTS (GNU/Linux 3.13.0-85-generic x86_64)
+
+ * Documentation:  https://help.ubuntu.com/
+
+  System information as of Wed May  4 21:45:43 UTC 2016
+
+  System load:  0.61              Processes:           88
+  Usage of /:   3.5% of 39.34GB   Users logged in:     0
+  Memory usage: 7%                IP address for eth0: 10.0.2.15
+  Swap usage:   0%
+
+  Graph this data and manage this system at:
+    https://landscape.canonical.com/
+
+  Get cloud support with Ubuntu Advantage Cloud Guest:
+    http://www.ubuntu.com/business/services/cloud
+
+0 packages can be updated.
+0 updates are security updates.
+```
+
+Or something as brief as this in the case of Ubuntu:
+
+```
+Welcome to Ubuntu 14.04.1 LTS (GNU/Linux 4.4.0-21-generic x86_64)
+
+ * Documentation:  https://help.ubuntu.com/
+```
+
+Congrats, you're now inside the Zulip dev environment! 
+
+You can confirm this by looking at the command prompt, which starts with
+`(zulip-venv)`. 
+
+Next you'll navigate to the shared Zulip directory `/srv/zulip`, activate the
+python venv, and start the web server:
+
+``` 
+(zulip-venv)vagrant@vagrant-ubuntu-trusty-64:~
+$ cd /srv/zulip 
+(zulip-venv)vagrant@vagrant-ubuntu-trusty-64:
+~$ source /srv/zulip-venv/bin/activate 
+(zulip-venv)vagrant@vagrant-ubuntu-trusty-64:
+~$ ./tools/run-dev.py --interface='' 
+```
+
+You will see several lines of output starting with something like:
+
+``` 
+2016-05-04 22:20:33,895 INFO: process_fts_updates starting
+Recompiling templates
+2016-05-04 18:20:34,804 INFO: Not in recovery; listening for FTS updates
+done
+Validating Django models.py...
+System check identified no issues (0 silenced).
+
+Django version 1.8
+Tornado server is running at http://localhost:9993/
+Quit the server with CTRL-C.
+2016-05-04 18:20:40,716 INFO     Tornado loaded 0 event queues in 0.001s
+2016-05-04 18:20:40,722 INFO     Tornado  95.5% busy over the past  0.0 seconds
+Performing system checks...
+``` 
+And ending with something similar to:
+
+``` 
+http://localhost:9994/webpack-dev-server/
+webpack result is served from http://localhost:9991/webpack/
+content is served from /srv/zulip
+
+webpack: bundle is now VALID.
+2016-05-06 21:43:29,553 INFO     Tornado  31.6% busy over the past 10.6 seconds
+2016-05-06 21:43:35,007 INFO     Tornado  23.9% busy over the past 16.0 seconds
+``` 
+
+Now the Zulip web server should be running and accessible. Verify this by
+navigating to [http://localhost:9991/](http://localhost:9991/) in your browser
+on your main machine.
```

Comment:
Maybe this would be a great place to put a screenshot of what the development environment login page looks like?